src/structs/framework_error.rs

```diff
@@ -178,39 +192,14 @@ pub enum FrameworkError<'a, U, E> {
         #[derivative(Debug = "ignore")]
         framework: crate::FrameworkContext<'a, U, E>,
         /// The interaction in question
-        interaction: crate::ApplicationCommandOrAutocompleteInteraction<'a>,
+        interaction: &'a serenity::CommandInteraction,
     },
     // #[non_exhaustive] forbids struct update syntax for ?? reason
     #[doc(hidden)]
     __NonExhaustive(std::convert::Infallible),
 }
 
 impl<'a, U, E> FrameworkError<'a, U, E> {
-    /// Returns the [`serenity::Context`] of this error
-    pub fn serenity_context(&self) -> &'a serenity::Context {
-        match *self {
-            Self::Setup { ctx, .. } => ctx,
-            Self::EventHandler { ctx, .. } => ctx,
-            Self::Command { ctx, .. } => ctx.serenity_context(),
-            Self::SubcommandRequired { ctx } => ctx.serenity_context(),
-            Self::CommandPanic { ctx, .. } => ctx.serenity_context(),
-            Self::ArgumentParse { ctx, .. } => ctx.serenity_context(),
-            Self::CommandStructureMismatch { ctx, .. } => ctx.serenity_context,
-            Self::CooldownHit { ctx, .. } => ctx.serenity_context(),
-            Self::MissingBotPermissions { ctx, .. } => ctx.serenity_context(),
-            Self::MissingUserPermissions { ctx, .. } => ctx.serenity_context(),
-            Self::NotAnOwner { ctx, .. } => ctx.serenity_context(),
-            Self::GuildOnly { ctx, .. } => ctx.serenity_context(),
-            Self::DmOnly { ctx, .. } => ctx.serenity_context(),
-            Self::NsfwOnly { ctx, .. } => ctx.serenity_context(),
-            Self::CommandCheckFailed { ctx, .. } => ctx.serenity_context(),
-            Self::DynamicPrefix { ctx, .. } => ctx.serenity_context,
-            Self::UnknownCommand { ctx, .. } => ctx,
-            Self::UnknownInteraction { ctx, .. } => ctx,
-            Self::__NonExhaustive(unreachable) => match unreachable {},
-        }
-    }
-
```

Comment:
Why was this method removed?

Reply:
We would have to match over the entire serenity FullEvent to get the context out of it, as is.

Reply:
Hmmm I see. Outright removing the method is really bad though for people who need it.

I think we should keep the method, but have it return `Option<&serenity::Context>`. It needs to be None one way or another because there may not be a serenity::Context available (FullEvent::Ratelimit). For now, it can just blanket return None if it would need to match over FullEvent. Later, when serenity gains a `FullEvent::context` method or similar, we can upgrade the method to work properly for those events that have an associated Context

I really don't want to remove the method outright without a designated replacement. You know, I'm thinking of the bot dev who couldn't care less about our API design and just wants his bot's features to keep working, and I don't want to pull the rug under those people by removing methods like these

Reply:
Is there not already the stopgap solution of calling `FrameworkError::ctx` then `Context::serenity_context`?

Reply:
Fixed by bumping to proper serenity 0.12, as the fullevent PR allowed this to exist again.